### ISSUE TITLE
单专家训练

### ISSUE THREAD

**Issue by IABAcademy** [lblaoke/TLC#11]
在论文Trustworthy Long-Tailed Classification中，Learning single expert这一部分描述了如何学习单个专家：
①对于证据和真值标签，计算负对数似然损失(用nll_loss实现)。
②对于所有预测失误的类对应的证据，将其与均匀分布进行KL散度计算
①能够诱导网络对正确分类产生更多证据；②能够使网络对错误分类产生更少证据。
现在我有两个问题：
首先，在我的一个长尾数据集中，我使用①能够获得与交叉熵损失近似的分类精度；然而，当我使用①+②学习单专家，无论②的权重有多小（我甚至设置过0.000001这样的权重），网络输出都近似于均匀分布，在调参上很困难。请问我该怎么做？
其次，我注意到，①和分类问题中常用的交叉熵损失相差无几；但从数值的角度来看，①+②使得网络的优化目标更类似对真值独热向量的预测精度，唯一的区别在于正确分类的数值上不封顶。在我对长尾数据集的实验中，我尝试将独热向量数值乘以一个倍数，之后使用MSE loss作为网络的损失函数，得到的结果和使用①+②学习单个专家几乎完全一致。而MSE loss并不适合分类问题。请问你认为，文章中提出的单专家学习方式，在实际网络训练中是否存在类似的风险？

**Comment by lblaoke** [comment]
关于②的问题，并不是只有预测失误的类才要算，应当是所有类的logits与均匀分布比较。②的权重是随着epoch而增长的，因为训练前期应当让模型优先提高分类精度，后期才需要用②来提高uncertainty estimation的效果。

**Comment by lblaoke** [comment]
关于MSE loss的问题，我并没有做过类似的实验，无法给出可靠的结论。但基于你的结果，我认为MSE可能也是一种可行的classification loss。在loss的选择上，NLL来自Multinomial分布，MSE来自Gaussian分布，而①+②来自Dirichlet分布。既然Multinomial和Dirichlet都可行，那么Gaussian可行也比较合理。
